containers/base/Dockerfile

```diff
@@ -116,27 +115,38 @@ RUN /tools/node/bin/npm install \
         node-cache@3.2.0 && \
     cd / && \
     /tools/node/bin/npm install -g forever && \
-    /tools/node/bin/npm install -g typescript
 
 # Install ProtoBuf, TensorFlow
-RUN pip install protobuf==3.0.0b2.post2 && \
+    pip install --no-cache-dir protobuf==3.0.0b2.post2 && \
     wget https://storage.googleapis.com/cloud-datalab/deploy/tf/tensorflow-0.11.0rc0-cp27-none-linux_x86_64.whl && \
-    pip install tensorflow-0.11.0rc0-cp27-none-linux_x86_64.whl && \
-    rm tensorflow-0.11.0rc0-cp27-none-linux_x86_64.whl
+    pip install --no-cache-dir tensorflow-0.11.0rc0-cp27-none-linux_x86_64.whl && \
+    rm tensorflow-0.11.0rc0-cp27-none-linux_x86_64.whl && \
 
 # Install DataFlow
-RUN pip install google-cloud-dataflow==0.4.2
+    pip install --no-cache-dir google-cloud-dataflow==0.4.2 && \
 
 # Install CloudML SDK
-RUN gsutil cp gs://cloud-ml/sdk/cloudml-0.1.6-alpha.tar.gz cloudml-0.1.6-alpha.tar.gz && \
-    pip install cloudml-0.1.6-alpha.tar.gz && \
-    rm cloudml-0.1.6-alpha.tar.gz
+    gsutil cp gs://cloud-ml/sdk/cloudml-0.1.6-alpha.tar.gz cloudml-0.1.6-alpha.tar.gz && \
+    pip install --no-cache-dir cloudml-0.1.6-alpha.tar.gz && \
+    rm cloudml-0.1.6-alpha.tar.gz && \
+
+# Clean up
+    apt-get purge -y build-essential bzip2 cpp cpp-4.9 python-setuptools pkg-config libfreetype6-dev && \
+    apt-get autoremove -y && \
+    rm -rf /var/lib/apt/lists/* && \
+    rm -rf /var/lib/dpkg/info/* && \
+    rm -rf /tmp/* && \
+    rm -rf /root/.cache/* && \
+    rm -rf /tools/google-cloud-sdk/.install && \
+    rm -rf /usr/share/doc/* && \
```

Comment:
I'm very concerned about the usability consequences of removing doc and man pages from a development tool.

How much space is that actually saving us?

(You also have "rm -rf /usr/share/doc/*" in here twice).